plugin/al/usdmayautils/AL/usdmaya/utils/DiffPrimVar.cpp

```diff
@@ -1674,6 +1715,78 @@ TfToken guessColourSetInterpolationTypeExtensive(
     return UsdGeomTokens->faceVarying;
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+TfToken guessColourSetInterpolationTypeExtensive(
+    const float*           rgba,
+    const size_t           numElements,
+    float                  threshold,
+    const size_t           numPoints,
+    MIntArray&             pointIndices,
+    MIntArray&             faceCounts,
+    std::vector<uint32_t>& indicesToExtract)
+{
+    // Specialized test if there is threshold provided.
+    // TODO: This function is extracted from guessColourSetInterpolationTypeExtensive() but without
+    // SSE/AVX optimization.
+    //       It is generally slower then the SSE/AVX version.
+    //       Improve it with SSE/AVX at some point when needed, for now the optimization is left for
+    //       the compiler.
+
+    if (numElements <= 1 || isWithinThreshold(rgba, numElements, threshold)) {
+        return UsdGeomTokens->constant;
+    }
+
+    float absThreshold = std::abs(threshold);
+
+    // check for per-vertex assignment
+    std::vector<uint32_t> indicesMap;
+    indicesMap.resize(numPoints, -1);
```

Comment:
You are using -1 here but 0xFFFFFFFF below. I would be clearfer if they were both the same. (Also, passing a negative value to initialize an unsigned integer without a cast might trigger a warning on some compilers and we generally have the warning-as-error compiler option turned on.)

Reply:
I have updated the code to check against the visited index, it is less memory friendly but more robust, let me know if that looks better?

Reply:
This is in the AL plugin, if you feel the change would impact performance, you can revert it. The callers are from data you control, if you feel confident the index are always valid by construction, feel free to remove the check, but I'd advise to do profiling first to verify if it really is a bottleneck.

Reply:
I reverted the changes in `DiffPrimVar.cpp`.
I have to admit that it was a copy-n-paste from `guessColourSetInterpolationTypeExtensive()` and the code has been used in production for a long time, I would keep it untouched in that case.